### ISSUE TITLE
Double check that our calculation of log_probs is correct and equivalent to PyTorch's distributions model

### ISSUE THREAD

**Issue by JoshVarty** [JoshVarty/HaliteIO#8]
Double check that our calculation of log_probs is correct and equivalent to PyTorch's distributions model

**Comment by JoshVarty** [comment]
Categorical implementation: 
https://github.com/pytorch/pytorch/blob/master/torch/distributions/categorical.py

DeepRL's Usage:
https://github.com/ShangtongZhang/DeepRL/blob/master/deep_rl/network/network_heads.py#L164-L190

**Comment by JoshVarty** [comment]
It appears to be the same. I took at look at an example from DeepRL's `CategoricalActorCritic` net with the following values

### DeepRL Values 
Logits
`[ -7.8911962e-06,  6.5093162e-05, -1.9086667e-05, -2.8031460e-05]`

Action
`[1]`

log_prob
`-1.3862`


### My Approach

softmax
`[0.2499974 , 0.25001564, 0.2499946 , 0.24999236]`

`log(0.25001564) = -1.3862318`


We seem to get the same values. So the moral of the story seems to be "Don't let your network output wildly different numbers". Perhaps better weight initalization?
